### ISSUE TITLE
Configuring anonymous key-auth plugin requires you to use UUIDs

### ISSUE THREAD

**Issue by davideagle** [Kong/kubernetes-ingress-controller#265]
NOTE: GitHub issues are reserved for **bug reports only**.
For anything else, please join the conversation
in Kong Nation https://discuss.konghq.com/c/kubernetes.

---

### Summary

Configuring anonymous key-auth plugin requires you to use UUIDs to reference Consumers which is not ideal

**Kong Ingress controller version**
0.3.0

**Kong or Kong Enterprise version**
Kong v1.1.0

### Kubernetes version

**paste `kubectl version` output**
v1.11.7

### What happened
Configuring anonymous key-auth plugin requires you to use UUIDs to reference Consumers

```yaml
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: wallet-openapi-key-auth
  namespace: default 
plugin: key-auth
config:
   anonymous: c97b40c3-56cf-11e9-8ba9-026db3b4995a
```
More Ideal way would be to use consumerRef in a similar way as you can in a KongCredential

```yaml
apiVersion: configuration.konghq.com/v1
kind: KongCredential
metadata:
  name: credential
consumerRef: consumer
type: acl
config:
  group: "group"
```

**Comment by Ngob** [comment]
Any plan on this matter? It make harder the integration with HELM (to configure an anonymous consumer on an auth plugin)

**Comment by hbagdi** [comment]
@Ngob Please see the above PR.

**Comment by hbagdi** [comment]
This will be fixed with Kong 1.4.
